src/custom-element.js

```diff
@@ -767,8 +767,12 @@ function createBaseAmpElementProto(win) {
     if (this.isInTemplate_) {
       return;
     }
-    this.resources_.remove(this);
-    this.implementation_.detachedCallback();
+    try {
+      this.implementation_.viewportCallback(false);
```

Comment:
This should instead:

1. Go inside `Resources.remove()` method and eventually call `Resource.setInViewport()`.
2. `Resource.setInViewport()` should only be called for a built elements.

It should also probably call `pauseCallback` as well. I know it's heavy and we can maybe make it lighter for this use case. But the concern I have is that detachment can happen due to reparenting and the further re-adding the element to DOM will have a resource in a wrong state that we can't recover. An example would be a video player that was playing. Reparenting will cause playback to pause, but the element will still "think" it's playing.